examples/svg_viewer/src/main.rs

```diff
@@ -0,0 +1,334 @@
+        constants: gfx::ConstantBuffer<Globals> = "Globals",
+    }
+}
+
+fn gpu_color(color: Color, opacity: f32) -> [f32; 4] {
+    [
+        f32::from(color.red) / 255.0,
+        f32::from(color.green) / 255.0,
+        f32::from(color.blue) / 255.0,
+        opacity
+    ]
+}
+
+// A very simple vertex constructor that only outputs the vertex position
+struct VertexCtor {
+    fill: Color
+}
+impl VertexConstructor<tessellation::FillVertex, GpuFillVertex> for VertexCtor {
+    fn new_vertex(&mut self, vertex: tessellation::FillVertex) -> GpuFillVertex {
+        assert!(!vertex.position.x.is_nan());
+        assert!(!vertex.position.y.is_nan());
+        GpuFillVertex {
+            // (ugly hack) tweak the vertext position so that the logo fits roughly
+            // within the (-1.0, 1.0) range.
+            position: (vertex.position * 0.0145 - vec2(1.0, 1.0)).to_array(),
+            color: gpu_color(self.fill, 1.0)
+        }
+    }
+}
+
+#[derive(Copy, Clone, Debug)]
+struct Scene {
+    zoom: f32,
+    pan: [f32; 2]
+}
+
+// extract the relevant globals from the scene struct
+impl From<Scene> for Globals {
+    fn from(scene: Scene) -> Self {
+        Globals {
+            zoom: [scene.zoom, scene.zoom],
+            pan: scene.pan
+        }
+    }
+}
+
+struct SvgPath {
+    d: String,
+    fill: Color
+}
+
+impl Default for SvgPath {
+    fn default() -> Self {
+        SvgPath {
+            d: String::new(),
+            fill: Color::new(0,0,0)
+        }
+    }
+}
+
+fn main() {
+    // Parse CLI arguments
+    let cli_args = App::new("Lyon CLI Renderer")
+                          .arg(Arg::with_name("input")
+                               .index(1)
+                               .value_name("FILE")
+                               .help("SVG file to render")
+                               .takes_value(true))
+                          .get_matches();
+
+    let filepath = cli_args.value_of("input").expect("Missing input file");
+
+    // read the SVG file into a string
+    let mut input_buffer = String::new();
+    let mut file = File::open(filepath)
+        .expect("Error opening input file!");
+    file.read_to_string(&mut input_buffer)
+        .expect("Error reading input file!");
+
+    // process the SVG tokens and generate an array of paths
+    let mut last_tag = None;
+    let mut svg_paths = Vec::new();
+    let mut current_path = SvgPath::default();
+    let mut tokens = Tokenizer::from_str(&input_buffer).tokens();
+    for token in &mut tokens {
+        match token {
+            // start of new tag
+            Token::SvgElementStart(tag) => {
+                last_tag = Some(tag);
+            }
+            // start of an SVG attribute
+            Token::SvgAttribute(name, value) => {
+                // ignore non-paths
+                if last_tag == Some(ElementId::Path) {
+                    match name {
+                        AttributeId::D => current_path.d = String::from(value.slice()),
```

Comment:
Not very important but you might as well just convert the string to a path directly here, and just manipulate path objects later.

Reply:
Nice one!  Converting to a path immediately also avoids allocating another `String` because the path builder takes a slice.